bpfd/src/command.rs

```diff
@@ -601,37 +545,81 @@ impl Program {
 
     pub(crate) fn if_index(&self) -> Option<u32> {
         match self {
-            Program::Xdp(p) => Some(p.info.if_index),
+            Program::Xdp(p) => p.if_index,
             Program::Tracepoint(_) => None,
-            Program::Tc(p) => Some(p.info.if_index),
+            Program::Tc(p) => p.if_index,
             Program::Kprobe(_) => None,
             Program::Uprobe(_) => None,
+            Program::Unsupported(_) => None,
+        }
+    }
+
+    pub(crate) fn set_if_index(&mut self, if_index: u32) {
+        match self {
+            Program::Xdp(p) => p.if_index = Some(if_index),
+            Program::Tc(p) => p.if_index = Some(if_index),
+            _ => (),
         }
     }
 
     pub(crate) fn if_name(&self) -> Option<String> {
         match self {
-            Program::Xdp(p) => Some(p.info.if_name.clone()),
+            Program::Xdp(p) => Some(p.iface.clone()),
             Program::Tracepoint(_) => None,
-            Program::Tc(p) => Some(p.info.if_name.clone()),
+            Program::Tc(p) => Some(p.iface.clone()),
             Program::Kprobe(_) => None,
             Program::Uprobe(_) => None,
+            Program::Unsupported(_) => None,
+        }
+    }
+
+    pub(crate) fn priority(&self) -> Option<i32> {
+        match self {
+            Program::Xdp(p) => Some(p.priority),
+            Program::Tracepoint(_) => None,
+            Program::Tc(p) => Some(p.priority),
+            Program::Kprobe(_) => None,
+            Program::Uprobe(_) => None,
+            Program::Unsupported(_) => None,
         }
     }
 
     pub(crate) fn direction(&self) -> Option<Direction> {
         match self {
             Program::Xdp(_) => None,
             Program::Tracepoint(_) => None,
-            Program::Tc(p) => Some(p.info.direction),
+            Program::Tc(p) => Some(p.direction),
             Program::Kprobe(_) => None,
             Program::Uprobe(_) => None,
+            Program::Unsupported(_) => None,
+        }
+    }
```

Comment:
ok we're getting a little carried away on providing accessors here.
if it's available in a majority of cases, cool.
but for cases like this where it's only applicable to one prog type we might as well make callers pattern match or if let.

e.g

```
if let Program::Tc(p) = program {
    p.direction
}
```